test/Microsoft.ML.Tests/Scenarios/Api/AspirationalExamples.cs

```diff
@@ -60,5 +60,98 @@ public void FirstExperienceWithML()
                 PetalLength = 5.1f
             });
         }
+
+
+        [Fact]
+        public void SimpleIrisDescisionTrees()
+        {
+            var env = new TlcEnvironment(new SysRandom(0), verbose: true);
+            string dataPath = "iris-data.txt";
+            // Create reader with specific schema.
+            var dataReader = TextLoader.CreateReader(env, ctx => (
+               Label: ctx.LoadText(0),
+               SepalWidth: ctx.LoadFloat(1),
+               SepalLength: ctx.LoadFloat(2),
+               PetalWidth: ctx.LoadFloat(3),
+               PetalLength: ctx.LoadFloat(4)),
+               dataPath);
+
+            // Load the data into the system.
+            var data = dataReader.Read(dataPath);
```

Comment:
>dataPath [](start = 39, length = 8)

this implies that our text reader will be an `IDataReader<string>`, right? Or `string` to be implicitly converted to `MultiFileSource`. I'm OK with the second idea, but is Tom?

Reply:
Hi @Zruty0 , I am OK with it I think. On multiple occassions I would have found an implicit conversion to `IMultiStreamSource` a useful simplifying instrument.

Reply:
I tried and failed. The implicit ctor is not called if you pass `dataPath` to a method: I guess it's only for assignments

---
In reply to: [215094774](https://github.com/dotnet/machinelearning/pull/809#discussion_r215094774) [](ancestors = 215094774)